backend/api/submissions/types.py

```diff
@@ -73,6 +72,12 @@ class Submission:
         LazyType["AudienceLevel", "api.conferences.types"]
     ] = restricted_field()
 
+    @strawberry.field
+    def notes(self, info) -> Optional[str]:
+        if CanSeeSubmissionPrivateFields().has_permission(self, info):
+            return self.notes
```

Comment:
I am thinking... do we want this behaviour for all `private_field`? maybe we could implement a generic solution inside `private_field` and `restricted_field`?

Reply:
Yeah probably, maybe in the future we want to show all the details on the Submission page if the user is an admin. Maybe when we are preparing the schedule...

Reply:
I was thinking to do something like this: 

```python
def private_field(field_name):
    """Field that can only be seen by admin and the submitter"""

    def resolver(self, info):
        if CanSeeSubmissionPrivateFields().has_permission(self, info):
            return getattr(self, field_name)
        return None

    return strawberry.field(resolver=resolver)


@strawberry.type
class Submission:
	...
    notes: Optional[str] = private_field("notes")
```

But I was thinking if there is a clever way to get the name of the field... and don't have to pass to the `private_filed` method... I'm pretty sure internally strawberry has to do something to retrieve it somehow...